emscripten.py

```diff
@@ -1480,10 +1480,27 @@ def create_receiving(function_table_data, function_tables_defs, exported_impleme
     receiving = '\n'.join(['var real_' + s + ' = asm["' + s + '"]; asm["' + s + '''"] = function() {''' + runtime_assertions + '''  return real_''' + s + '''.apply(null, arguments);
 };
 ''' for s in exported_implemented_functions if s not in ['_memcpy', '_memset', 'runPostSets', '_emscripten_replace_memory', '__start_module']])
+
+  module_exports = exported_implemented_functions + function_tables(function_table_data)
+
+  # Generate an exports file that records all the exported symbols from asm.js/wasm module.
+  module_exports_suppressions = '\n'.join(['/**\n * @suppress {duplicate, undefinedVars}\n */\nvar %s;\n' % i for i in module_exports])
+  exports_file = get_configuration().get_temp_files().get('_module_exports.js')
+  shared.Settings.MODULE_EXPORTS_FILE = exports_file.name
```

Comment:
I think we should avoid using a file for this. How about storing the list of wasm/asm exports in a plain list (`MODULE_EXPORTS`, without `_FILE`), and only if we run closure, we generate a file for it in the code about to run closure?

Reply:
Yeah, that's doable - adjusted.